py/dml/codegen.py

```diff
@@ -543,7 +543,10 @@ def expr_unop(tree, location, scope):
             report(WSIZEOFTYPE(tree.site))
             return codegen_sizeof(
                 tree.site, mkLit(tree.site, cident(var), None))
-    rh = codegen_expression_maybe_nonvalue(rh_ast, location, scope)
+    try:
+        rh = codegen_expression_maybe_nonvalue(rh_ast, location, scope)
+    except EIDENT as e:
+        raise ESIZEOFOP(e.site, e.identifier) if op == 'sizeof' else e
```

Comment:
unsure, but `else None` *might* be better; subtle differences between `raise` and `raise e`

Reply:
Huh! Didn't know about that feature of python.

Reply:
Oh, it's not. `raise None` is invalid, what you mean is `raise`: it would be rewritten as:
```
if op == 'sizeof':
    raise ESIZEOFOP(e.site, e.identifier)
raise
```

Reply:
OK. I recall seeing messages from Python talking about None being raised when referring to a naked `raise` statement. Maybe it was a py2 thing. Your suggestion is clearer in any case.